recipes/corrade/all/conanfile.py

```diff
@@ -58,24 +59,28 @@ def source(self):
     def _configure_cmake(self):
         if not self._cmake:
             self._cmake = CMake(self)
-            self._cmake.definitions["BUILD_STATIC"] = not self.options.shared
-            self._cmake.definitions["BUILD_DEPRECARED"] = self.options["build_deprecated"]
-            self._cmake.definitions["WITH_INTERCONNECT"] = self.options["with_interconnect"]
-            self._cmake.definitions["WITH_MAIN"] = self.options["with_main"]
-            self._cmake.definitions["WITH_PLUGINMANAGER"] = self.options["with_pluginmanager"]
-            self._cmake.definitions["WITH_TESTSUITE"] = self.options["with_testsuite"]
-            self._cmake.definitions["WITH_UTILITY"] = self.options["with_utility"]
-            self._cmake.definitions["WITH_RC"] = "ON"
+            def add_cmake_option(option, value):
+                var_name = "{}".format(option).upper()
+                value_str = "{}".format(value)
+                var_value = "ON" if value_str == 'True' else "OFF" if value_str == 'False' else value_str 
+                self._cmake.definitions[var_name] = var_value
+                print("{0}={1}".format(var_name, var_value))
+
+            for option, value in self.options.items():
+                add_cmake_option(option, value)
+
+            add_cmake_option("BUILD_STATIC", not self.options.shared)
+            add_cmake_option("BUILD_STATIC_PIC", not self.options.shared and self.options.get_safe("fPIC") == True)
+
+            if self.settings.compiler == "Visual Studio":
+                self._cmake.definitions["MSVC2015_COMPATIBILITY"] = "ON" if self.settings.compiler.version == "14" else "OFF"
+                self._cmake.definitions["MSVC2017_COMPATIBILITY"] = "ON" if self.settings.compiler.version == "15" else "OFF"
+                self._cmake.definitions["MSVC2019_COMPATIBILITY"] = "ON" if self.settings.compiler.version == "16" else "OFF"
```

Comment:
```suggestion
                self._cmake.definitions["MSVC2015_COMPATIBILITY"] = self.settings.compiler.version
                self._cmake.definitions["MSVC2017_COMPATIBILITY"] = self.settings.compiler.version
                self._cmake.definitions["MSVC2019_COMPATIBILITY"] = self.settings.compiler.version == "16"
```

Reply:
What happened to `== "14"` and `== "15"`?

Reply:
Also I could've reused `add_cmake_option(...)` here as well.

Reply:
again, don't need an extra method for a very simple logic.